app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SessionDataMissing.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscriptionfrontend.controllers
+
+import play.api.Logger
+import play.api.mvc.{Result, Results}
+
+trait SessionDataMissing {
+  this: Results =>
+
+  def showSessionDataMissing(): Result = {
```

Comment:
I don't think the name of this method reflects what it does. How about 'redirectIfSessionMissing'?